apps/glusterfs/app.go

```diff
@@ -291,6 +291,11 @@ func (a *App) SetRoutes(router *mux.Router) error {
 			Method:      "POST",
 			Pattern:     "/devices/{id:[A-Fa-f0-9]+}/state",
 			HandlerFunc: a.DeviceSetState},
+		rest.Route{
```

Comment:
The model for device state changes is to use `/devices/{id..}/state`. It is then handled by the Node and Device state changes as seen [in node_entry.go](https://github.com/heketi/heketi/blob/master/apps/glusterfs/node_entry.go#L248), where the value that should be used to fail a node or device is `api.EntryStateFailed`.